Libraries/Text/RCTShadowText.m

```diff
@@ -306,6 +306,11 @@ - (void)_setParagraphStyleOnAttributedString:(NSMutableAttributedString *)attrib
     CGFloat lineHeight = round(_lineHeight * fontSizeMultiplier);
     paragraphStyle.minimumLineHeight = lineHeight;
     paragraphStyle.maximumLineHeight = lineHeight;
+    // vertically center text
+    CGFloat fontSize = round(_fontSize * (_allowFontScaling && self.fontSizeMultiplier > 0.0 ? self.fontSizeMultiplier : 1.0));
+    [attributedString addAttribute:NSBaselineOffsetAttributeName
+                             value:[NSNumber numberWithFloat:lineHeight/2 - fontSize/2]
```

Comment:
Nit: can you replace

    [NSNumber numberWithFloat:lineHeight/2 - fontSize/2]

With

    @(lineHeight/2 - fontSize/2)